ProjectBuildingException due to dependency with `<packaging>bundle</packaging>`

As soon as the `spdx-maven-plugin` hits a dependency with `<packaging>bundle</packaging>` a ProjectBuildingException is raised.
The bundle packaging is part of the OSGi support, see https://github.com/xerial/snappy-java/issues/255

To reproduce add dependency `org.xerial.snappy:snappy-java:1.1.10.5` to your project and try to run the goal `spdx:createSPDX`.

```
--- spdx:1.0.0-RC3-SNAPSHOT:createSPDX (default-cli) @ *removed* ---
spdx file type = .json
Creating SPDX File C:\*removed*.spdx.json
Null package file
Maven Project Build Error creating SPDX package for dependency artifact ID snappy-java
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
Unknown packaging: bundle @ line 6, column 16

    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:208)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:333)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:295)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.createSpdxPackage (SpdxV2DependencyBuilder.java:421)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:190)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:216)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:216)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.CreateSpdxMojo.buildSpdxDependencyInformation (CreateSpdxMojo.java:815)
    at org.spdx.maven.CreateSpdxMojo.execute (CreateSpdxMojo.java:618)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.model.building.ModelBuildingException: 1 problem was encountered while building the effective model for org.xerial.snappy:snappy-java:1.1.10.5
Unknown packaging: bundle @ line 6, column 16

    at org.apache.maven.model.building.DefaultModelProblemCollector.newModelBuildingException (DefaultModelProblemCollector.java:176)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:550)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:419)
    at org.apache.maven.model.building.DefaultModelBuilder.build (DefaultModelBuilder.java:248)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:176)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:333)
    at org.apache.maven.project.DefaultProjectBuilder.build (DefaultProjectBuilder.java:295)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.createSpdxPackage (SpdxV2DependencyBuilder.java:421)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:190)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:216)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.utils.SpdxV2DependencyBuilder.addMavenDependency (SpdxV2DependencyBuilder.java:216)
    at org.spdx.maven.utils.AbstractDependencyBuilder.addMavenDependencies (AbstractDependencyBuilder.java:89)
    at org.spdx.maven.CreateSpdxMojo.buildSpdxDependencyInformation (CreateSpdxMojo.java:815)
    at org.spdx.maven.CreateSpdxMojo.execute (CreateSpdxMojo.java:618)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Error creating SPDX package for dependency artifact ID snappy-java.  A minimal SPDX package will be created.
```

--- a/src/it/simple-bundle-it/pom.xml
+++ b/src/it/simple-bundle-it/pom.xml
@@ -1,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>org.spdx.it</groupId>
+  <artifactId>simple-bundle-it</artifactId>
+  <version>1.0-SNAPSHOT</version>
+
+  <description>A simple IT which includes a dependency with bundle packaging.</description>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.release>11</maven.compiler.release>
+  </properties>
+
+  <dependencies>
+    <dependency>
+      <groupId>commons-collections</groupId>
+      <artifactId>commons-collections</artifactId>
+      <version>3.2.2</version>
+    </dependency>
+    <dependency>
+      <groupId>org.xerial.snappy</groupId>
+      <artifactId>snappy-java</artifactId>
+      <version>1.1.10.5</version>
+    </dependency>
+    <dependency>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <version>4.13.2</version>
+      <scope>test</scope>
+    </dependency>
+  </dependencies>
+
+  <build>
+    <plugins>
+      <plugin>
+        <groupId>@project.groupId@</groupId>
+        <artifactId>@project.artifactId@</artifactId>
+        <version>@project.version@</version>
+        <executions>
+          <execution>
+            <id>build-spdx</id>
+            <goals>
+              <goal>createSPDX</goal>
+            </goals>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <version>3.11.0</version>
+      </plugin>
+    </plugins>
+  </build>
+</project>

--- a/src/it/simple-bundle-it/src/main/java/simple/Simple.java
+++ b/src/it/simple-bundle-it/src/main/java/simple/Simple.java
@@ -1,0 +1,24 @@
+/*
+ * Copyright 2014 Source Auditor Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package simple;
+
+public class Simple
+{
+    public static void main(String[] args)
+    {
+        System.out.println( "Hello World!");
+    }
+}

--- a/src/it/simple-bundle-it/src/test/java/SimpleTest.java
+++ b/src/it/simple-bundle-it/src/test/java/SimpleTest.java
@@ -1,0 +1,27 @@
+/*
+ * Copyright 2014 Source Auditor Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package simple;
+
+import org.junit.Test;
+
+public class SimpleTest
+{
+    @Test
+    public void test()
+    {
+
+    }
+}

--- a/src/it/simple-bundle-it/verify.groovy
+++ b/src/it/simple-bundle-it/verify.groovy
@@ -1,0 +1,3 @@
+File spdxFile = new File( basedir, "target/site/org.spdx.it_simple-bundle-it-1.0-SNAPSHOT.spdx.json" );
+
+assert spdxFile.isFile()

--- a/src/main/java/org/spdx/maven/utils/SpdxV2DependencyBuilder.java
+++ b/src/main/java/org/spdx/maven/utils/SpdxV2DependencyBuilder.java
@@ -422,6 +422,7 @@ public class SpdxV2DependencyBuilder
         try
         {
             ProjectBuildingRequest request = new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+            request.setProcessPlugins( false );
             request.setRemoteRepositories( mavenProject.getRemoteArtifactRepositories() );
             for ( ArtifactRepository ar : request.getRemoteRepositories() ) {
                 LOG.debug( "request Remote repository ID: {}", ar.getId() );

--- a/src/main/java/org/spdx/maven/utils/SpdxV3DependencyBuilder.java
+++ b/src/main/java/org/spdx/maven/utils/SpdxV3DependencyBuilder.java
@@ -420,6 +420,7 @@ public class SpdxV3DependencyBuilder
         try
         {
             ProjectBuildingRequest request = new DefaultProjectBuildingRequest( session.getProjectBuildingRequest() );
+            request.setProcessPlugins( false );
             request.setRemoteRepositories( mavenProject.getRemoteArtifactRepositories() );
             for ( ArtifactRepository ar : request.getRemoteRepositories() ) {
                 LOG.debug( "request Remote repository ID: {}", ar.getId() );

